src/txreconciliation.cpp

```diff
@@ -69,10 +100,56 @@ class TxReconciliationTracker::Impl {
         return std::make_tuple(we_initiate_recon, we_respond_recon, RECON_VERSION, m_local_recon_salt);
     }
 
+    bool RegisterPeer(NodeId peer_id, bool peer_inbound,
+        bool they_may_initiate, bool they_may_respond, uint32_t recon_version, uint64_t remote_salt)
+    {
+        // We do not support reconciliation salt/version updates. We treat an attempt to update
+        // after a successful registration as a protocol violation.
+        LOCK(m_mutex);
+        if (m_states.find(peer_id) != m_states.end()) return false;
```

Comment:
This check means that duplicated `SENDRECON` are processed as protocol violation, correct ?

I wonder if we have Bitcoin nodes widely deployed over non-TCP stack, where the packet dedup is not guaranteed. I guess that should be so rare a case, though if we learn some and the duplication rate is high enough to trigger this reconciliation failure we can still relax this check.

Reply:
Yeah good point, keeping it open for now so that maybe others comment if they want to relax the check.

Reply:
Bitcoin P2P connections are stateful, so you can't generally run them over say a UDP stack anyway.